src/source_sender/mod.rs

```diff
@@ -74,11 +85,13 @@ impl SourceSender {
             buf_size: CHUNK_SIZE,
             inner: None,
             named_inners: Default::default(),
+            lag_time: Some(register_histogram!(LAG_TIME_NAME)),
```

Comment:
Only caveat is that the call to `SourceSender::builder` needs to happen once we're inside the span for the source component. As long as that's happening, the registered metric will have the right metric labels from the span fields, etc.

Reply:
Part of this PR _is_ to run the builder inside the source component span.

Reply:
Right you are. I did not review closely enough. :)